README.md

```diff
@@ -214,6 +214,8 @@ curl_easy_setopt(curl_handle, CURLOPT_CAINFO, "/etc/pki/tls/certs/ca-bundle.crt"
    - You can try hitting the non-TLS version of the endpoint if available. (Not Recommended).
 1. **No known conversion between `std::string` and `Aws::String`**
     - Either turn off custom memory management in the AWS C++ SDK or build it as a static library (`-DBUILD_SHARED_LIBS=OFF`)
+1. **I'm getting wierd linking errors like `/usr/bin/ld: /home/<...>bin/lib/libaws-lambda-runtime.a: error adding symbols: file format not recognized clang: error: linker command failed with exit code 1 (use -v to see invocation)`**
+   - By default this project enables _link-time optimization (LTO)_ (configurable with `ENABLE_LTO` _cmake_ option). This tells compilers to delay optimizations until the linking phase. For some compilers (like _clang_) this means that each compilation unit should be compiled down only to intermediate representations (_e.g._ _clang_ uses _LLVM IR_) instead of directly to native object files. As a side-effect, library archives (like _aws-lambda-runtime_) would also contain only _IR_ code. This may result in linker errors like the above if you use a linker incompatible with the _IR_ code generated by your compiler (e.g. _GNU ld_ linker doesn't support _LLVM IR_). In order for the linking step to succeed with _LTO_ you must either use an appropriate linker with something like `-DCMAKE_CXX_FLAGS="-fuse-ld=lld"` (_GNU ld_ for _gcc_ and _lld_ for _clang_ are known to work) or disable _LTO_ (`-DENABLE_LTO=OFF`).
```

Comment:
>this means that each compilation unit should be compiled down only to intermediate representations (_e.g._ _clang_ uses _LLVM IR_) instead of directly to native object files.

Moreover, I'd rather avoid making such claims on how the linker/compiler works because:
- I'm not an expert so I really don't know.
- If that information gets outdated in the future, we don't end up with stale or wrong claims.

I think it's better to suggest turning off LTO and keep it succinct. That's assuming the problem exists, which I'm not yet sure is that case.

Reply:
Totally agreeded. You're right. I'm gonna try to rephrase that tomorrow. That kind of turned out too assertive.

Reply:
Thank you!
Also, it would be nice if you can provide me with steps on triggering this problem. I can't reproduce it.

Reply:
Working on it

Reply:
Done. It's on the issue #151